NEWS.md

```diff
@@ -5,6 +5,11 @@ Main
 -------
 ### Features
 
+### DYAMOND-summer initial conditions
+
+Added functionality to allow initial conditions to be overwritten by 
+interpolated datasets. Currently supports equilibrium moisture models. 
```

Comment:
Is the intent for the function added to be general, or is it only for the dyamond case?

Reply:
cc @sriharshakandala Do we wish to maintain this `overwrite` method for all such future initial condition file-reads? Or is there some way this can be done entirely within the `LocalState` prescription to avoid such an `overwrite` ?

Reply:
With this workflow, the user can define a custom overwrite function for each of the supported initial condition types if, for example, they wish to use data from a file for setting up the initial condition.
Currently, this is primarily intended for the `DYAMONDSummer` simulation which reads initial condition from file. We are planning to refactor the initial condition prescription to separate out
1. Allocation of prognostic variables and
2. initialization of prognostic variables
We plan to address this refactoring in a different PR.

Reply:
Generalized to handle `filepath`s by @Sbozzolo